packages/components/addon/components/hds/code-block/index.hbs

```diff
@@ -0,0 +1,30 @@
+{{!
+  Copyright (c) HashiCorp, Inc.
+  SPDX-License-Identifier: MPL-2.0
+}}
+
+<div class={{this.classNames}} ...attributes>
+  <div class="hds-code-block__header">
+    {{~yield (hash Title=(component "hds/code-block/title"))~}}
+    {{~yield (hash Description=(component "hds/code-block/description"))~}}
+  </div>
+
+  <pre
+    class="hds-code-block__code"
+    {{style maxHeight=@maxHeight}}
+    contentEditable={{not this.isReadOnly}}
+    data-line={{@highlightLines}}
+    id={{if @hasCopyButton this.preCodeId null}}
```

Comment:
it's safe to always apply the DOM ID, and we remove a cycle.
```suggestion
    id={{this.preCodeId}}
```

Reply:
Updated in 07e385bc16fc8cbcc9030d5a5083b5b3a499ca8c